internal/app/handler/shortener.go

```diff
@@ -0,0 +1,107 @@
+package handler
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"math/rand"
+	"net/http"
+
+	"github.com/FoGezz/go-url-shortener/cmd/shortener/config"
+	"github.com/FoGezz/go-url-shortener/internal/app/storage"
+	"github.com/go-chi/chi/v5"
+)
+
+var alphabet []rune = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+type ShortenerHandler struct {
+	http.Handler
+	linksContainer *storage.LinksContainer
+	cfg            config.Config
+}
+
+type postShortenHandler struct {
+	ShortenerHandler
+}
+
+func NewPostShortenHandler(container *storage.LinksContainer, cfg config.Config) *postShortenHandler {
+	s := &postShortenHandler{} //у меня не получилось здесь сразу передать linksContainer
```

Comment:
Why?

Reply:
Ok, I came up with this form
```	
func NewPostShortenHandler(linksMapping *storage.LinksMapping, cfg *config.Config) *postShortenHandler {
	s := postShortenHandler{
		ShortenerHandler: ShortenerHandler{
			storage: linksMapping,
			cfg: cfg,
		},
	}

	return &s
```

But TBH it is less clear that previous direct assignment